### ISSUE TITLE
Symbols inside `TYPE_CHECKING` are importable from other files

### ISSUE THREAD

**Issue by afdaniele** [microsoft/pyright#11001]
## Environment data

-   Pylance version: 2025.8.3
-   OS and version: Ubuntu 22.04.5 LTS
-   Python version (& distribution if applicable, e.g. Anaconda): `3.10.12` (system interpreter)

## Code Snippet

```python
# foo.py

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    import numpy as np


# bar.py

from foo import np

np.abs()
```

## Repro Steps

1. Create a `foo.py` with the content above;
2. Create a `bar.py` with the content above;

## Expected behavior

In `bar.py`, we should not be able to import `np` via `foo.py` because it is in the `TYPE_CHECKING` block. 

## Actual behavior

Pylance is perfectly happy with the import and shows a healthy Numpy docs bubble if I hover over `np` in `bar.py` (see image below).
The worst part is that on projects with multiple files and classes, VSCode suggests importing symbols via indirect import paths that pass through `TYPE_CHECKING` blocks.

<img width="2297" height="660" alt="Image" src="https://github.com/user-attachments/assets/bd57bd72-059a-4533-81fd-a3880aaec99f" />

**Comment by erictraut** [comment]
Pyright's behavior here is by design.

From the perspective of a type checker, code within a `TYPE_CHECKING` block is treated like any other code and is not special-cased. The `TYPE_CHECKING` symbol is effectively a way to "lie" to a type checker, and there are many reasons this may be done. In general, I recommend avoiding the use of `TYPE_CHECKING` if possible because it causes the type checker to analyze code that is different from what is seen at runtime.
